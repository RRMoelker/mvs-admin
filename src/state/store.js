// import { reduxSwarmLogMiddleware } from '@philholden/redux-swarmlog';
import { combineReducers, createStore, compose, applyMiddleware } from 'redux';
import { createAction } from 'redux-actions';
import moment from 'moment';

import { timeMiddleware } from './middleware.js';
import challengeReducer from './challenge/reducer';
import timerReducer, { TIMER_SET } from './timer/reducer';

// import keys from '../keys.json';
// import { addReduxSwarmLog, configureReduxSwarmLog } from '@philholden/redux-swarmlog';

// addReduxSwarmLog({
//     name: 'mvs-admin',
//     keys
// });

const rootReducer = combineReducers({
    timer: timerReducer,
    challenge: challengeReducer
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const enhancer = composeEnhancers(
    applyMiddleware(
        timeMiddleware,
        // reduxSwarmLogMiddleware
    )
);
const initialState = {};
const store = createStore(rootReducer, initialState, enhancer);

// configureReduxSwarmLog({ reduxStore: store });

export default store;
