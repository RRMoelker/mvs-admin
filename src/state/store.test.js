import store from './store.js';

it('should initialize the store', () =>{
    const state = store.getState();
    expect(state).toMatchSnapshot();
});
