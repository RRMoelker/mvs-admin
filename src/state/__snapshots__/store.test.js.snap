// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should initialize the store 1`] = `
Object {
  "challenge": Object {
    "list": Array [],
  },
  "timer": Object {
    "running": true,
    "time": 0,
  },
}
`;
